RCAL-1168: Update COSMOS fluxes using `roman_photoz` simulated catalog.

mairanteodoro opened the pull request:
Resolves [RCAL-1168](https://jira.stsci.edu/browse/RCAL-1168)

This PR implements functionality that updates the observed fluxes of galaxies generated by `romanisim` with the scaled ones generated by `roman_photoz`. This option is activated by providing the `--use-photoz` argument in the CLI.

Now `generate_input_catalog.InputCatalog` accepts an optional `flux_catalog_filename` argument that points to a flux catalog produced by `roman_photoz`. When provided, the generated `romanisim` catalog will be updated using the supplied flux catalog instead of attempting to run `roman_photoz` internally.

This PR is connected to another PR in `roman_photoz`:
- https://github.com/spacetelescope/roman_photoz/pull/42

mairanteodoro committed:
Remove option to disable scaling.

mairanteodoro committed:
Add observation plan example file.

schlafly commented on roman_simulate_dr/scripts/generate_input_catalog.py:
```diff
@@ -22,49 +26,73 @@ def __init__(
+            f"with open({str(flux_path)!r}, 'wb') as f:\n"
+            "    pickle.dump(updated, f)\n"
+        )
+        subprocess.run([sys.executable, "-c", script], check=True)
```
If you make the flux_catalog an argument instead, can we avoid this?  i.e., make it so that this option requires an additional filename input that points to the flux_catalog output from roman_photoz?

mairanteodoro replied:
Yes, the code has been update to do just that.

mairanteodoro committed:
Update roman-photoz dependency to use main branch

mairanteodoro committed:
Refactor to stop running roman-photoz internally.

This class now accepts an optional flux_catalog_filename argument that
points to a flux catalog produced by roman_photoz. When provided, the
generated romanisim catalog will be updated using the supplied flux
catalog instead of attempting to run roman_photoz internally.

mairanteodoro committed:
Add roman-photoz simulated file for unit tests.

mairanteodoro committed:
Prevent running tests using python 3.14.

schlafly reviewed (approved):
Thanks, this looks good.
